src/Workspaces/Core/Portable/CodeGeneration/CodeGenerationHelpers.cs

```diff
@@ -38,7 +37,8 @@ internal static class CodeGenerationHelpers
             return factory.ThrowStatement(exceptionCreationExpression);
         }
 
-        public static TSyntaxNode AddAnnotationsTo<TSyntaxNode>(ISymbol symbol, TSyntaxNode syntax)
+        [return: NotNullIfNotNull("syntax")]
```

Comment:
`syntax` is constrained to `SyntaxNode`, so it can never be substituted for a nullable node type. Perhaps `where TSyntaxNode : SyntaxNode?`